my-app/src/components/login/LogInComponent.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
+import authService from "../../services/auth.service";
+import { Redirect } from "react-router-dom";
+
+class LogIn extends Component {
+  state = {
+    isAuthorized: authService.isAuthorized(), // on init
+    error: null,
+    isLoaded: false,
+    username: "",
+    password: "",
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    authService
+      .login({
+        username: this.state.username,
+        password: this.state.password,
+      })
+      .then((res) => {
+        this.setState({ isAuthorized: authService.isAuthorized() });
+      })
+      .catch((error) => {
+        alert(error.message);
+      });
+  };
+
+  onChangeInput = (e) => {
+    const name = e.target.name;
+    this.setState({ [name]: e.target.value });
+  };
+
+  render() {
+    const { isAuthorized } = this.state;
+    // = const isAuthorized = this.state.isAuthorized;
+
+    return isAuthorized ? (
+      <Redirect to="/departments" />
```

Comment:
Dont you use the protected routes for this?

Reply:
You know I've thought about it and the thing is it's a Login page and we don't have to login twice that's why there is this check...

Reply:
Okay? I'll try to figure it out)